meson.build

```diff
@@ -74,7 +74,8 @@ if iconv_dep.found() or cc.has_function('iconv_open')
     conf.set('CONFIG_ICONV', 1)
 endif
 
-freetype_dep = dependency('freetype2', version: '>= 2.10.92', default_options: ['default_library=' + get_option('default_library')]) # workaround for https://github.com/mesonbuild/meson/issues/8047
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
+                          default_options: ['default_library=' + get_option('default_library'), 'harfbuzz=disabled']) # workaround for https://github.com/mesonbuild/meson/issues/8047
```

Comment:
fyi, freetype without HarfBuzz has limited (no?) auto-hinting. Not sure if regular hinting is also affected.
This doesn't really matter for actual ASS *(as hinting should be disabled for this anyway)* but may be desired for converted subs.

Reply:
freetype can only be built against freetype-enabled harfbuzz or no harfbuzz, so if the issue here is that meson subprojects can't deal with circular dependencies, then it probably can't be resolved without adding a temporary subproject or so *(not sure if worth the effort; what's the convention for the feature-set of meson subproject/fallback builds?)*

Reply:
It seems to have built against non-freetype enabled harfbuzz? Not sure if that will cause issues in practice.

I personally don't think it's worth the effort to try and hack around the circular dependency issue in a downstream library, and would rather either take a swing at a proper fix in meson or just say "if you use the subprojects you won't have hinting working correctly".

Reply:
freetype is [using `<hb-ft.h>` if Harfbuzz-support](https://gitlab.freedesktop.org/freetype/freetype/-/blob/master/src/autofit/afshaper.h#L30) was enabled. This harfbuzz-header contains functions only implemented when Harfbuzz was built with freetype support. It may be possible to circumvent this in static builds (since before linking you only need the header), but it should fail in linking if harfbuzz doesn't have the functions *(or does it stub them out?)*.
Can you link the resulting static lib(s) successfully?

If there's no easy way to resolve this and meson-subproject-builds aren't typically expected to have all/most functionality, I think it's fine to leave it disabled.

Reply:
I think the CI build is statically linked? https://github.com/libass/libass/pull/330/files#diff-3539c292179efe94f388bc73d93594ba5bada7a7a41f5b2bf273957185de7fd4R22

Logs indicate it uses both the harfbuzz and freetype subprojects... dunno. I can either disable it again or just assume that if it builds we're fine, up to you really. I'm leaving it disabled for Aegisub CI regardless.

Reply:
Yeah, then it'll never reach linking, so if meson just uses the headers from harfbuzz source, the error will not show itself during building. The logs also claim Harfbuzz was configured without freetype support.
Can you try a CI-build with the `--enable-test`-equivalent set, so it actually tries to link the lib?